tests/generate_small_sample.py

```diff
@@ -4,113 +4,160 @@
 
 from pysource_minimize import minimize
 import sys
-import subprocess as sp
 import textwrap
 import os
 import linecache
 from executing import Source
-from multiprocessing import Pool,get_context
+from multiprocessing import get_context
 import tempfile
 import hashlib
 import time
+import contextlib
+import os
+from rich.progress import Progress, track
+from rich.syntax import Syntax
+from rich.console import Console
+import argparse
 
 last_samples_dir = Path(__file__).parent / "last_samples"
 last_samples_dir.mkdir(exist_ok=True)
-samples_dir = Path(__file__).parent / "samples"
 
 
-# big_samples = (Path(__file__).parent / "samples").glob("*.py")
-# big_samples = list(Path.home().rglob("*.py"))
 small_samples = Path(__file__).parent / "small_samples"
 
 
-def big_samples():
+def source_hash(source_code):
+    return hashlib.sha256(source_code.encode("utf8")).hexdigest()
+
+
+def big_samples(folder):
     yield from last_samples_dir.rglob("*.py")
-    yield from samples_dir.rglob("*.py")
 
     hashes = set()
-    root = Path.home()
-    # root = Path(__file__).parent / "samples"
-    for p in root.rglob("*.py"):
+
+    for p in folder.rglob("*.py"):
         try:
             content = p.read_text()
         except:
             continue
 
-        if content.count("\n") > 100000:
+        if content.count("\n") > 50000:
+            # long take to much time to check and are most likely generated code or repetitive
             continue
 
-        h = hashlib.sha256(content.encode()).hexdigest()
+        h = source_hash(content)
         if h in hashes:
             continue
         hashes.add(h)
         yield p
 
 
 def test_file(filename: Path):
-    sys.setrecursionlimit(3000)
-
-    if filename.read_text().count("\n") > 50000:
-        return True, filename
+    code = filename.read_text()
 
-    # clear caches
+    # Clear caches because we are analysing a lot of bytecode here and this can result in hight memory usage.
+    # This is usually not a problem for executing, but this usage scenario is different
     linecache.clearcache()
     for cache_name in ("__source_cache_with_lines", "__executing_cache"):
         if hasattr(Source, cache_name):
             delattr(Source, cache_name)
 
     test = TestFiles()
     try:
-        test.check_filename(filename, check_names=True)
+        with open(os.devnull, "w") as dev_null:
+            with contextlib.redirect_stderr(dev_null):
+                with contextlib.redirect_stdout(dev_null):
+                    test.check_filename(filename, check_names=True)
```

Comment:
```suggestion
        with (
            open(os.devnull, "w") as dev_null,
            contextlib.redirect_stderr(dev_null),
            contextlib.redirect_stdout(dev_null)
        ):
            test.check_filename(filename, check_names=True)
```

This became possible in 3.10

Reply:
but I want to use this for older python versions too (3.6 maybe)